web/src/lib/domain-logic/essentialQuestions.ts

```diff
@@ -0,0 +1,155 @@
+interface EssentialQuestion
+  extends Omit<EssentialQuestionsProps<ProfileData[keyof ProfileData]>, "fieldName" | "choices"> {
+  fieldName: keyof IndustrySpecificData;
+  validationFunc: EssentialQuestionFunction;
+  reset: boolean;
+  convenienceFunc: (industryId: string | undefined) => boolean;
+}
+
+export const getResetIndustrySpecificData = (industryId: string | undefined) => {
+  const industry = LookupIndustryById(industryId);
+  return EssentialQuestions.filter((eQ) => {
+    return eQ.reset && !eQ.validationFunc(industry);
+  }).reduce((reducer, eQ) => {
+    return { ...reducer, [eQ.fieldName]: emptyIndustrySpecificData[eQ.fieldName] };
+  }, {} as Partial<IndustrySpecificData>);
+};
+
+export const getValidationFuncByFieldName = (fieldName: ProfileLabelField) => {
+  return EssentialQuestions.find((eQ) => {
+    eQ.contentFieldName == fieldName || eQ.fieldName == fieldName;
+  })?.validationFunc;
+};
+
+export const validateByIndustryIdAndFieldName = (fieldName: ProfileLabelField) => {
+  return (industryId: string | undefined): boolean => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return EssentialQuestions.find((eQ) => {
+      return eQ.fieldName == fieldName || eQ.contentFieldName == fieldName;
+    })!.validationFunc(LookupIndustryById(industryId));
+  };
+};
+
+export const hasEssentialQuestion = (industryId: string | undefined): boolean => {
+  const industry = LookupIndustryById(industryId);
+  const essentialQuestionResults = EssentialQuestions.map((essentialQuestionFunction) => {
+    return essentialQuestionFunction.validationFunc(industry);
+  });
+  return essentialQuestionResults.includes(true);
+};
+
+export const isInterstateLogisticsApplicable = validateByIndustryIdAndFieldName("interstateLogistics");
+export const isInterstateMovingApplicable = validateByIndustryIdAndFieldName("interstateMoving");
+export const isCarServiceApplicable = validateByIndustryIdAndFieldName("carService");
+export const isCertifiedInteriorDesignerApplicable =
+  validateByIndustryIdAndFieldName("certifiedInteriorDesigner");
+export const isChildcareForSixOrMoreApplicable = validateByIndustryIdAndFieldName("isChildcareForSixOrMore");
+export const isCpaRequiredApplicable = validateByIndustryIdAndFieldName("requiresCpa");
+export const isCannabisLicenseApplicable = validateByIndustryIdAndFieldName("cannabisLicenseType");
+export const isLiquorLicenseApplicable = validateByIndustryIdAndFieldName("liquorLicense");
+export const isProvidesStaffingServicesApplicable =
+  validateByIndustryIdAndFieldName("providesStaffingService");
+export const isRealEstateAppraisalManagementApplicable = validateByIndustryIdAndFieldName(
+  "realEstateAppraisalManagement"
+);
+
+export const EssentialQuestions: EssentialQuestion[] = [
+  {
+    reset: true,
+    validationFunc: (industry) => {
+      return (
+        !!industry.industryOnboardingQuestions.isInterstateTransportApplicable && industry.id === "logistics"
```

Comment:
I don't think we need `!!` on all of these because they are already booleans, right? You only need that if you're going to type-coerce something like a string into a boolean type. But things like `industry.industryOnboardingQuestions.isInterstateTransportApplicable` should already be a boolean

Reply:
a few of them can be resolved as undefined, so the double bang is needed to avoid a typescript error